dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/DefaultEventChangeLogService.java

```diff
@@ -52,6 +53,20 @@ public class DefaultEventChangeLogService implements EventChangeLogService {
   private final HibernateTrackedEntityDataValueChangeLogStore trackedEntityDataValueChangeLogStore;
   private final TrackerAccessManager trackerAccessManager;
 
+  @Transactional
+  @Override
+  public void addEventChangeLog(
```

Comment:
Are we always calling this with either a `dataElement` or an `eventProperty` (the other arg being null)? If so providing two methods make a clearer API. The repo underneath can stay the same. Having many string args like this is awkward in terms of programmer errors. If you put it in the wrong order you got a bug.